hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsAsyncQueueService.kt

```diff
@@ -0,0 +1,120 @@
+package uk.gov.justice.hmpps.sqs
+
+import com.google.gson.GsonBuilder
+import com.google.gson.ToNumberPolicy
+import com.microsoft.applicationinsights.TelemetryClient
+import org.slf4j.LoggerFactory
+import software.amazon.awssdk.services.sqs.SqsAsyncClient
+import software.amazon.awssdk.services.sqs.model.DeleteMessageRequest
+import software.amazon.awssdk.services.sqs.model.GetQueueAttributesRequest
+import software.amazon.awssdk.services.sqs.model.Message
+import software.amazon.awssdk.services.sqs.model.QueueAttributeName
+import software.amazon.awssdk.services.sqs.model.ReceiveMessageRequest
+import software.amazon.awssdk.services.sqs.model.SendMessageRequest
+import kotlin.coroutines.suspendCoroutine
+import kotlin.math.min
+import software.amazon.awssdk.services.sqs.model.PurgeQueueRequest as AwsPurgeQueueRequest
+
+open class HmppsAsyncQueueService(
+  private val telemetryClient: TelemetryClient?,
+  hmppsAsyncTopicFactory: HmppsAsyncTopicFactory,
+  hmppsAsyncQueueFactory: HmppsAsyncQueueFactory,
+  hmppsSqsProperties: HmppsSqsProperties,
+) {
```

Comment:
Now the async and sync clients are completely unrelated we've had to duplicate all of the sync client helpers for async (e.g. HmppsAsyncQueueService) - I'm hoping to remove this duplication in future PRs but we'll see how it goes. Walking that fine line between DRY and inscrutable abstractions.